source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -70,12 +78,63 @@
             CurrentPosition: this.currentPosition,
             Seniority: this.seniority,
             FullName: this.fullName,
-            Period: this.period
+            Period: this.period,
+            IsValid: this.isValid()
         };
     }
-
     var viewmodel = new PeriodCreation();
 
+    $('#generate-evaluation-button').on('click', function () {
+        if (viewmodel.isValid()) {
+            postEvaluationsForGeneration({
+                onSuccess: function (result) {
+                    showEvaluationGenerationResult(true, result.amountGenerated);
+                },
+                onError: function () {
+                    alert('Fallo interno. Por favor recargue la página.');
+                }
+            });
+        } else {
+            showEvaluationGenerationResult(false);
+        }
+
+        function postEvaluationsForGeneration(callbacks) {
+            var model = viewmodel.toJs();
+            var modelFiltered = { Employees: _.filter(model.Employees, function (e) { return e.ResponsibleId && !e.Period; }) };
+            $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
+                type: "POST",
+                dataType: 'json',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify(modelFiltered),
+                complete: function (response) {
+                    var result = { amountGenerated: parseInt(response.responseText) }
+                    if (typeof callbacks.onSuccess === 'function') { callbacks.onSuccess(result); }
+                }
+            });
+            error = callbacks.onError;
+        };
+
+        function showEvaluationGenerationResult(wereGenerated, amountGenerated) {
+            if (wereGenerated) {
+                var modalContainer = $('#evaluations-generated-confirm');
+                modalContainer.find('.modal-title').text("EVALUACIONES GENERADAS");
+                var countText = (amountGenerated == '1') ? "Se ha generado 1 evaluación correctamente" : "Se han generado " + amountGenerated + " evaluaciones correctamente";
+                modalContainer.find('.modal-text').text(countText);
+                modalContainer.find('.back').hide();
+                modalContainer.find('.confirm').show();
+                modalContainer.modal('show');
+            } else {
+                var modalContainer = $('#evaluations-generated-confirm');
+                modalContainer.find('.modal-title').text("EVALUACIONES NO GENERADAS");
+                var text = "No se puedieron generar las evaluaciones, datos incorrectos!";
+                modalContainer.find('.modal-text').text(text);
+                modalContainer.find('.back').show();
+                modalContainer.find('.confirm').hide();
+                modalContainer.modal('show');
+            }
```

Comment:
You've got a lot of repeated code between both branches of the `if`. Also, you can make use of the boolean variable to simplify it.

```javascript
var modalContainer = $('#evaluations-generated-confirm'); // remember to prefer classes to IDs
// prefer ? for conditional expressions
modalContainer.find('.modal-title').text(wereGenerated ? "Evaluaciones generadas" : "Evaluaciones no generadas");

// prefer ? for conditional expressions
var text = wereGenerated
    ? amountGenerated == '1' ? "Se ha generado 1 evaluación correctamente" : "Se han generado " + amountGenerated + " evaluaciones correctamente";
    : "No se pudieron generar las evaluaciones, datos incorrectos!"; // TYPO: "Puedieron"

modalContainer.find('.modal-text').text(text);  // common to if and else
// jQuery toggle: show for true, hide for false
modalContainer.find('.back').toggle(!wereGenerated); 
modalContainer.find('.confirm').toggle(wereGenerated);
modalContainer.modal('show'); // common to if and else
```